### PR TITLE
Remove Docker/kubeadm dependency

### PR DESCRIPTION
**Pull request by klausenbusk**
Less dependencies are better!

### CODE REVIEW

File: scripts/prepare_cluster.sh [hunk 1]
```diff
@@ -7,14 +7,16 @@ source $VARIABLES
 if [ -z $MASTER_CERTIFICATE_KEY ]
 then
     echo "[info] MASTER_CERTIFICATE_KEY not set. Generating new."
-    MASTER_CERTIFICATE_KEY=$(docker run kubeadocker alpha certs certificate-key)
+    # https://github.com/kubernetes/kubernetes/blob/cde45fb161c5a4bfa7cfe45dfd814f6cc95433f7/cmd/kubeadm/app/phases/copycerts/copycerts.go#L80-L87
+    MASTER_CERTIFICATE_KEY=$(head -c32 /dev/random | sha256sum | awk '{print $1}')
```
Comment (skovbakke) [inline scripts/prepare_cluster.sh:11]: Why exactly do you compute the `sha256sum`?

Comment (klausenbusk) [reply scripts/prepare_cluster.sh:11]: We need a string, `/dev/random` provides bytes.

Comment (skovbakke) [reply scripts/prepare_cluster.sh:11]: So it is just a convenient way of turning the bytes into a hex encoded string.
Now why do we care to specify picking 32 bytes from random, the sha256sum always outputs 32 bytes regardless of the input length. 

Maybe we should be more transparent and convert it to hex directly?
```suggestion
    MASTER_CERTIFICATE_KEY=$(head -c32 /dev/random | hexdump -e '"%x"')
```

Comment (klausenbusk) [reply scripts/prepare_cluster.sh]: Changed to `MASTER_CERTIFICATE_KEY=$(hexdump -e '"%x"' /dev/random | head -c64)`, sounds good?

Comment (skovbakke) [reply scripts/prepare_cluster.sh]: yes

**Review by skovbakke — DISMISSED** [review]
LGTM 😀

**Comment by skovbakke** [comment]
Needs review

**Review by jacobcrawford — DISMISSED** [review]
LGTM

**Pull request merged**